### PR TITLE
Modified everflow script to run on Marvell chip and enabling everflow support for T0-52 topology

### PR DESCRIPTION
**Pull request by rajneeshaec**
### Description of PR
Modified the everflow script to support it on marvell asic

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [-/ ] Test case(new/improvement)

### Back port request
- [-/ ] 201911
- [-/ ] 202012
- [ -/] 202205

#### What is the motivation for this PR?
Modified the everflow script to support it on marvell asic

#### How did you verify/test it?
test suite is passing on community testbed

#### Any platform specific information?
Yes(related to marvell asic)

### CODE REVIEW

File: tests/everflow/test_everflow_ipv6.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 from everflow_test_utilities import setup_info  # noqa: F401, E501 lgtm[py/unused-import] pylint: disable=import-error
 
 pytestmark = [
-    pytest.mark.topology("t0", "t1", "m0")
+    pytest.mark.topology("t0", "t1", "m0", "t0-52")
```
Comment (Blueve) [inline tests/everflow/test_everflow_ipv6.py:12]: t0-52 is included in topo t0, no need to add it specifically.

File: tests/everflow/test_everflow_testbed.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
 from tests.common.fixtures.ptfhost_utils import copy_arp_responder_py # noqa: F401, E501 lgtm[py/unused-import] pylint: disable=import-error
 
 pytestmark = [
-    pytest.mark.topology("t0", "t1", "m0")
+    pytest.mark.topology("t0", "t1", "m0", "t0-52")
```
Comment (Blueve) [inline tests/everflow/test_everflow_testbed.py:18]: t0-52 is included in topo t0, no need to add it specifically.

File: tests/everflow/test_everflow_ipv6.py (hunk 1 above)
Comment (rajneeshaec) [reply tests/everflow/test_everflow_ipv6.py]: addressed comment

File: tests/everflow/test_everflow_testbed.py (hunk 2 above)
Comment (rajneeshaec) [reply tests/everflow/test_everflow_testbed.py]: addressed comment

File: tests/everflow/everflow_test_utilities.py [hunk 3]
```diff
@@ -725,6 +725,8 @@ def get_expected_mirror_packet(mirror_session, setup, duthost, mirror_packet, ch
         expected_packet.set_do_not_care_scapy(packet.IP, "len")
         expected_packet.set_do_not_care_scapy(packet.IP, "flags")
         expected_packet.set_do_not_care_scapy(packet.IP, "chksum")
+        if duthost.facts["asic_type"] == 'marvell':
```
Comment (Blueve) [inline tests/everflow/everflow_test_utilities.py:728]: might need consider marvell-armhf

Comment (rajneeshaec) [reply tests/everflow/everflow_test_utilities.py:728]: Hi @Blueve

I have checked the ansible facts gathered by the code. facts asic_type is marvell and asic_subtype is marvell-armf. These changes are common for all marvell devices. I think we should use asic_type=marvell here

Please find details given below:-

ansible_facts": {"discovered_interpreter_python": "/usr/bin/python", "dut_basic_facts": {"asic_subtype": "marvell-armhf", "asic_type": "marvell", "branch": "202205", "build_date": "Thu Nov 24 12:44:20 UTC 2022", "build_number": 179361, "build_version": "202205.179361-7c1015b45

Thanks
Rajneesh

**Comment by Blueve** [comment]
@rajneeshaec there is a conflicts in your PR, we can merge your PR after you resolved the conflict. Thanks ; )

**Comment by rajneeshaec** [comment]
Hi @Blueve

   I have raised a new pull request for this. Could you please review it and merge it 

https://github.com/sonic-net/sonic-mgmt/pull/7247

Thanks
Rajneesh

**Pull request closed without merging**